packages/main/src/RPA/DocumentAI/Base64AI.py

```diff
@@ -298,3 +308,86 @@ def get_user_data(self) -> Dict:
+        query image.
+        """
+        # NOTE(cmin764): There's no mock support for this API.
+        recognize_endpoint = self._to_endpoint("signature/recognize")
+
+        payload = {
+            "reference": reference_image,
+            "query": query_image,
+        }
+        for key, value in list(payload.items()):
+            value, is_url = self._url_or_file(value)
+            key += "Url" if is_url else "Image"
+            payload[key] = value
+        del payload["reference"]
+        del payload["query"]
+
+        response = requests.post(
+            recognize_endpoint,
+            headers=self._request_headers,
+            json=payload,
+        )
+        response.raise_for_status()
+        match_response = response.json()
+        return match_response
+
+    def filter_matching_signatures(
+        self,
+        match_response: JSONType,
+        confidence_threshold: float = 0.8,
+        similarity_threshold: float = 0.8,
+    ) -> Dict[Tuple[int, Tuple[int, ...]], List[Dict[str, Any]]]:
+        """Gets through all the recognized signatures in the queried image and returns
+        only the ones passing the confidence thresholds.
+        """
+        accepted_references = [
+            True if ref["confidence"] >= confidence_threshold else False
+            for ref in match_response["reference"]
+        ]
+        candidates = {}
+        # As (left, top, right, bottom) tuple.
+        to_coords = lambda item: (
+            item["left"],
+            item["top"],
+            item["left"] + item["width"],
+            item["top"] + item["height"],
+        )
+
+        for qry_idx, candidate in enumerate(match_response["query"]):
+            if candidate["confidence"] < confidence_threshold:
+                continue  # not sure enough it is a signature
+
+            qry_coords = to_coords(candidate)
+            for ref_idx, similarity in enumerate(candidate["similarities"]):
+                if (
+                    not accepted_references[ref_idx]
+                    or similarity < similarity_threshold
+                ):
+                    # Skip any resemblance to doubtful reference signatures.
+                    # Skip not similar enough signatures.
+                    continue
+
+                # Filtered properties for the accepted match.
+                qry_body = {
+                    "index": qry_idx,
+                    "coords": qry_coords,
+                    "similarity": similarity,
+                }
+                # Every accepted reference signature (index, coordinates) contains a
+                #  list of similar recognized signatures from the queried image, as
+                #  {index, coordinates, similarity}.
+                ref_coords = to_coords(match_response["reference"][ref_idx])
+                # Example of such candidate dictionary entry:
+                #  (0, (92, 509, 234, 616)): [{'index': 0, 'coords': (1812, 813, 2114, 1035), 'similarity': 0.89}]
```

Comment:
The parsed dictionary of matching signatures will look like so:
<img width="849" alt="Screenshot 2023-02-07 at 15 28 28" src="https://user-images.githubusercontent.com/709053/217191983-01bc9d80-fcb9-4c6a-a243-8eebf5727838.png">

If this is not self-explanatory, then we should simplify the structure. (keep in mind that the returned structure will also be documented so can be understood)